test/testmatrix.cpp

```diff
@@ -537,11 +537,40 @@ TEST(Matrix, luDecompPivoting)
     x = A.solve(rhs);
 
     CHECK_EQUAL((4*c - 3*b)/(a*c), x(0));
-    CHECK_EQUAL(3*(b - c)/c, x(1));
+    CHECK_EQUAL(-(3*(-b + c)/c), x(1));
```

Comment:
Regarding the writing of a unit test of the pivot elements. The matrix after e.g. A.solve() is not change on is it? In the solvedChecked function in matrix.cpp. You do a copy of the Matrix at " Matrix PLU(*this);"? or is this just a reference?